feat(config): support user configured env (#27028)
renovatebot/renovate: docs/usage/self-hosted-configuration.md

## allowScripts

## allowedEnv

> Users can specify a set of allowed environment variable names through `allowedEnv`, and only those specified in the list will be accepted in the [`env`](./configuration-options.md#env) configuration.

Reviewer: ```suggestion
Users can set allowed environment variable names through `allowedEnv`.
Only the environment variables on the list will be accepted in the [`env`](./configuration-options.md#env) configuration.
```

Maybe something like this is easier to read and understand.

## allowScripts

## allowedEnv

Bot administrators can allow users to configure custom environment variables within repo config.
Only the environment variables on the list will be accepted in the [`env`](./configuration-options.md#env) configuration.

Examples:

```json title="renovate.json"
{
  "env": {
    "SOME_ENV_VARIABLE": "some_value",
    "EXTRA_ENV_NAME": "value"
  }
}
```

The above would require `allowedEnv` to be configured similar to the following:

```js title="config.js"
module.exports = {
  allowedEnv: ['SOME_ENV_*', 'EXTRA_ENV_NAME'],
};
```

> 

Reviewer 2: ```suggestion

`allowedEnv` values can be either:

 - An exact match string (e.g. `EXTRA_ENV_NAME`), or
 - A minimatch pattern (e.g. `SOME_ENV_*`), or
 - A regex expression (e.g. `/_ENV$/`).

```

## allowScripts

## allowedEnv

Bot administrators can allow users to configure custom environment variables within repo config.
Only environment variables matching the list will be accepted in the [`env`](./configuration-options.md#env) configuration.

Examples:

```json title="renovate.json"
{
  "env": {
    "SOME_ENV_VARIABLE": "some_value",
    "EXTRA_ENV_NAME": "value"
  }
}
```

The above would require `allowedEnv` to be configured similar to the following:

```js title="config.js"
module.exports = {
  allowedEnv: ['SOME_ENV_*', 'EXTRA_ENV_NAME'],
};
```

`allowedEnv` values can be exact match header names, glob patterns, or regex patterns.

> For more details on the syntax and supported patterns, see Renovate's [String Pattern Matching documentation](https://docs.renovatebot.com/string-pattern-matching/).

Reviewer: This link should be changed to a relative link, so that Material for MkDocs can warn us if the link needs to be changed later (if we rename the linked file for example).

## allowScripts

## allowedEnv

Bot administrators can allow users to configure custom environment variables within repo config.
Only environment variables matching the list will be accepted in the [`env`](./configuration-options.md#env) configuration.

Examples:

```json title="renovate.json"
{
  "env": {
    "SOME_ENV_VARIABLE": "some_value",
    "EXTRA_ENV_NAME": "value"
  }
}
```

The above would require `allowedEnv` to be configured similar to the following:

```js title="config.js"
module.exports = {
  allowedEnv: ['SOME_ENV_*', 'EXTRA_ENV_NAME'],
};
```

`allowedEnv` values can be either:

- An exact match string (e.g. `EXTRA_ENV_NAME`), or
- A minimatch pattern (e.g. `SOME_ENV_*`), or
- A regex expression (e.g. `/_ENV$/`).

> 

Reviewer 2: Replace with link to new docs page once https://github.com/renovatebot/renovate/pull/27387 is merged
